components/app.js

```diff
@@ -81,4 +85,50 @@ class App {
 	handleDeleteGradesSuccess() {
 		this.getGrades();
 	}
+	reviseGrade(id) {
+		document.getElementById('labelAdd').textContent = 'Update a Grade';
+		var submit = document.querySelector('button[type="submit"]');
+		submit.textContent = 'Update';
+		currentID = id;
+	}
+	editGrade(name, course, grade) {
+		var data;
+		if (name && !course && !grade) {
```

Comment:
This conditional is _**very**_ manual. Rather than having such a long series of checks, you should make an empty object before the conditional, and have an individual conditional checking if each of the parameters has a value, and if it does, add that property to your object.